packages/ui/src/components/atoms/Dropdown.tsx

```diff
@@ -0,0 +1,156 @@
+  overflow-y: auto;
+  z-index: 10;
+  &:hover {
+    background-color: #272320;
+  }
+`;
+
+const ListItem = styled.li`
+  background-color: #46403c;
+  &:hover {
+    background-color: #272320;
+  }
+  /* margin-left: 40px; */
+`;
+
+const Container = styled.div<{ isOpen: boolean }>`
+  position: relative;
+  width: 100%;
+  border-radius: 8px;
+  background-color: #272320;
+  input {
+    padding-right: 30px;
+    border-color: ${props => (props.isOpen ? '#D4AF37' : '#272320')};
+  }
+  &:hover {
+    input {
+      border-color: #d4af37;
+    }
+    ${List} {
+      background-color: #191715;
+      cursor: pointer;
+    }
+  }
+`;
+
+const IconContainer = styled.div`
+  position: absolute;
+  right: 24px;
+  top: 60%;
+  transform: translateY(-50%);
+`;
+
+interface DropdownProps {
+  items: LeanBoxProps[];
+  shouldChangeColor?: boolean;
+}
+
+export const Dropdown: React.FC<DropdownProps> = ({
+  items,
+  shouldChangeColor,
+}) => {
+  const [search, setSearch] = useState('');
+  const [selectedItem, setSelectedItem] = useState<LeanBoxProps | null>(null);
+  const [isOpen, setIsOpen] = useState(false);
+  const [hoveredItemId, setHoveredItemId] = useState<any | null>(null);
+
+  const filteredItems = items.filter(item =>
+    item.text.toLowerCase().includes(search.toLowerCase()),
+  );
+
+  const handleInputChange = (value: string) => {
+    setSearch(value);
+  };
+
+  const toggleDropdown = () => {
+    setIsOpen(!isOpen);
+    setSearch('');
+  };
+
+  const handleItemSelection = (item: LeanBoxProps) => {
+    setSelectedItem(item);
+    toggleDropdown();
+  };
+
+  return (
+    <Container isOpen={isOpen}>
+      {selectedItem ? (
+        <TempListItem
+          {...selectedItem}
+          onClick={toggleDropdown}
+          restrictedItem
+          shouldChangeColor={shouldChangeColor}
+          leftImageSrc={
+            shouldChangeColor ? cypherockRedIcon : selectedItem?.leftImageSrc
+          }
+        />
+      ) : (
+        <Input
+          type="text"
+          value={search}
+          name="choose"
+          onClick={toggleDropdown}
+          onChange={handleInputChange}
+          bgColor="#272320"
+          placeholder={isOpen ? 'Search' : 'Choose a coin'}
+        />
+      )}
+      <IconContainer>
+        <Image
+          src={isOpen ? searchIcon : triangleInverseIcon}
+          alt={isOpen ? 'Search Icon' : 'Dropdown Icon'}
+        />
+      </IconContainer>
+
+      {isOpen && (
+        <List onMouseLeave={toggleDropdown}>
+          {filteredItems.map(item => (
+            <ListItem
+              key={item.id}
+              onClick={() => handleItemSelection(item)}
+              onMouseEnter={() => setHoveredItemId(item.id)}
+              onMouseLeave={() => setHoveredItemId(null)}
+            >
+              <TempListItem
+                {...item}
+                selectedItem={selectedItem}
+                isHovered={item.id === hoveredItemId}
+              />
+            </ListItem>
```

Comment:
Why do we need to pass `isHovered`? The `TempListItem` should be able to handle its own hover effect. Also, is there a need to use `ListItem`, can just `TempListItem` do the job?

Reply:
TempListIteam, even this I will try to change the name but its style does not match the style of dropdownlist. Hence ListItem Style.
IsHovered, will change

Reply:
isHovered is removed completely now, hover state is handled in DropDownListItem